### PR TITLE
Convert Banner styles to vanilla-extract

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Subj

## Justify why they are needed

Refactor before changing banner-related logic in next PR

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4137?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4138** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4138?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4137** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4137?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4136** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4136?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4135** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4135?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/Banner/Banner.css.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { style, styleVariants } from '@vanilla-extract/css'
+import { colors, minWidth, theme } from 'ui/src/theme'
+
+const GridArea = {
+  Content: 'content',
+  CloseBtn: 'close-btn',
+}
+
+export const bannerRoot = styleVariants({
```
Comment (guilhermespopolin) [inline apps/store/src/components/Banner/Banner.css.ts:9]: Nice. Much cleaner ✨

**Review by guilhermespopolin — APPROVED** [review]

File: apps/store/src/components/Banner/Banner.tsx [hunk 2]
```diff
@@ -17,15 +18,15 @@ type Props = {
 
 export const Banner = ({ children, handleClose, variant = 'info' }: Props) => {
   return (
-    <Wrapper variant={variant}>
-      <Content>
-        <StyledIcon variant={variant} />
+    <div className={clsx(bannerRoot.base, bannerRoot[variant])}>
```
Comment (Youakeem) [inline apps/store/src/components/Banner/Banner.tsx:21]: Isn't `base` shared between all variants?

Comment (alebedev) [reply apps/store/src/components/Banner/Banner.tsx:21]: Not automagically.  We could've specified composition in `css.ts`, but that's more boilerplate.  Something like this example from vanilla docs: https://vanilla-extract.style/documentation/api/style-variants/#style-composition

I think that clsx solution is both smaller and easier to understand

**Pull request merged**